roboflow/models/object_detection.py

```diff
@@ -142,11 +153,28 @@ def predict(
         else:
             self.__exception_check(image_path_check=image_path)
 
+        resize = False
         # If image is local image
         if not hosted:
             if type(image_path) is str:
                 image = Image.open(image_path).convert("RGB")
                 dimensions = image.size
+                original_dimensions = copy.deepcopy(dimensions)
+
+                # Here we resize the image to the preprocessing settings before sending it over the wire
+                if "resize" in self.preprocessing.keys():
+                    if dimensions[0] > int(
+                        self.preprocessing["resize"]["width"]
+                    ) or dimensions[1] > int(self.preprocessing["resize"]["height"]):
+                        image = image.resize(
```

Comment:
Looks like this only does `Stretch To` but should follow the same resize strat they preproc'd with.

Reply:
^^ @paulguerrie when we get a chance let's add img_dims back to classification response in infer server, that way we can know for hosted urls - @Tyler-Odenthal fixed for local classifications